src/cli/commands/login.command.ts

```diff
@@ -162,4 +208,44 @@ export class LoginCommand {
             return Response.error(e);
         }
     }
+
+    private async getSsoCode(codeChallenge: string, state: string): Promise<string> {
+        return new Promise((resolve, reject) => {
+            const callbackServer = http.createServer((req, res) => {
+                const urlString = 'http://localhost' + req.url;
+                const url = new URL(urlString);
+                const code = url.searchParams.get('code');
+                const receivedState = url.searchParams.get('state');
+                res.setHeader('Content-Type', 'text/html');
+                if (code != null && receivedState != null && receivedState === state) {
+                    res.writeHead(200);
+                    res.end('<html><head><title>Success | Bitwarden CLI</title></head><body>' +
+                        '<h1>Successfully authenticated with the Bitwarden CLI</h1>' +
+                        '<p>You may now close this tab and return to the terminal.</p>' +
+                        '</body></html>');
+                    callbackServer.close(() => resolve(code));
+                } else {
+                    res.writeHead(400);
+                    res.end('<html><head><title>Failed | Bitwarden CLI</title></head><body>' +
+                        '<h1>Something went wrong logging into the Bitwarden CLI</h1>' +
+                        '<p>You may now close this tab and return to the terminal.</p>' +
+                        '</body></html>');
+                    callbackServer.close(() => reject());
+                }
+            });
+            const webUrl = this.environmentService.webVaultUrl == null ? 'https://vault.bitwarden.com' :
+                this.environmentService.webVaultUrl;
+            for (let port = 8065; port <= 8070; port++) {
+                try {
+                    this.ssoRedirectUri = 'http://localhost:' + port;
+                    callbackServer.listen(port, async () => {
+                        await open(webUrl + '/#/sso?clientId=cli' +
+                            '&redirectUri=' + encodeURIComponent(this.ssoRedirectUri) +
+                            '&state=' + state + '&codeChallenge=' + codeChallenge);
+                    });
+                } catch { }
+                break;
```

Comment:
Shouldn't `break;` of the loop be inside of the `try { }` portion of this, otherwise it'll try exactly once (assuming the idea here is to try against several ports until one successfully is open/available?)

Reply:
yea, this looks wrong. will fix.

Reply:
fixed

Reply:
Also, wouldn't you also need to call the `reject` if an open port could not be established or code execution reaches the end of this method? Perhaps `return;` would appropriate instead of `break;` where with outside of the loop finally rejecting the promise to resolve/return control to the main flow.